tests/translate/storage/test_php.py

```diff
@@ -335,6 +335,29 @@ def test_parsing_arrays_keys_with_quotes(self):
 
         assert bytes(phpfile).decode() == phpsource
 
+    def test_parsing_arrays_keys_with_number_as_value(self):
+        """Ensure that our identifiers can have numbers as value. Bug #5355."""
+        phpsource = r"""<?php
+return [
+    1 => 'value1',
+    2 => 'value2',
+    '3' => 'value3',
```

Comment:
Let's include some additional quotes to verify escaping actually works:

```suggestion
    '3\'' => 'value3',
    '4"' => "value4",
    '5\'' => "value4",
```